extensions/open-with-app/src/index.tsx

```diff
@@ -47,10 +48,11 @@ function ApplicationsListItem(props: { application: Application }) {
             onAction={async () => {
               try {
                 const selectedItems = await getSelectedFinderItems();
+                await closeMainWindow({ clearRootSearch: true, popToRootType: PopToRootType.Suspended });
                 if (selectedItems.length != 0) {
-                  selectedItems.forEach((item) => {
-                    open(item.path.replace(/"/g, '\\"'), application.bundleId);
-                  });
+                  for (let i = 0; i < selectedItems.length; i++) {
+                    await open(selectedItems[i].path.replace(/"/g, '\\"'), application.bundleId);
```

Comment:
We should remove these last leftovers from the old code. This call could be simplified to:

```js
await open(selectedItems[i].path, application);
```